<?xml version="1.0" encoding="UTF-8"?>
<pages xmlns="http://jboss.com/products/seam/pages"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="http://jboss.com/products/seam/pages http://jboss.com/products/seam/pages-2.2.xsd"

       no-conversation-view-id="/home.xhtml"
       login-view-id="/login/login.xhtml">

    <page view-id="*" login-required="true">
        <navigation> 
            <rule if-outcome="menuProduto">
            	<end-conversation before-redirect="true" root="true"/>
                <redirect view-id="/cadastros/produto.xhtml"/>
            </rule>
            <rule if-outcome="menuUsuario">
            	<end-conversation before-redirect="true" root="true"/>
                <redirect view-id="/cadastros/usuarioFiltro.xhtml"/>
            </rule>
            <rule if-outcome="menuLeilao">
            	<end-conversation before-redirect="true" root="true"/>
                <redirect view-id="/leilao/leilaoFiltro.xhtml"/>
            </rule>                        
   		</navigation> 
    </page>

    <exception class="org.jboss.seam.framework.EntityNotFoundException">
        <redirect view-id="/erro/error.xhtml">
            <message severity="warn">Record not found</message>
        </redirect>
    </exception>
    
    <exception class="javax.persistence.EntityNotFoundException">
        <redirect view-id="/erro/error.xhtml">
            <message severity="warn">Record not found</message>
        </redirect>
    </exception>
    
    <exception class="javax.persistence.EntityExistsException">
        <redirect view-id="/erro/error.xhtml">
            <message severity="warn">Duplicate record</message>
        </redirect>
    </exception>
    
    <exception class="javax.persistence.OptimisticLockException">
        <end-conversation/>
        <redirect view-id="/erro/error.xhtml">
            <message severity="warn">Another user changed the same data, please try again</message>
        </redirect>
    </exception>
    
    <exception class="org.jboss.seam.security.AuthorizationException">
        <redirect view-id="/erro/error.xhtml">
            <message severity="error">Você não tem permissão para acessar esta página ou recurso</message>
        </redirect>
    </exception>
    
    <exception class="org.jboss.seam.security.NotLoggedInException">
        <redirect view-id="/login/login.xhtml">
            <message severity="warn">#{messages['org.jboss.seam.NotLoggedIn']}</message>
        </redirect>
    </exception>
    
    <exception class="javax.faces.application.ViewExpiredException">
        <redirect view-id="/login/login.xhtml">
            <message severity="warn">Your session has timed out, please try again</message>
        </redirect>
    </exception>
    
    <exception class="org.jboss.seam.ConcurrentRequestTimeoutException" log-level="trace">
      <http-error error-code="503" />
    </exception>
     
    <exception>
        <redirect view-id="/erro/error.xhtml">
            <message severity="error">Unexpected error, please try again</message>
        </redirect>
    </exception>
    
</pages>
